### PR TITLE
Add climate platform to Matter docs

### PR DESCRIPTION
**Pull request by marcelveldt**
## Proposed change
Add climate platform to Matter docs

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/95434
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/matter.markdown [hunk 1]
```diff
@@ -16,6 +16,7 @@ ha_codeowners:
 ha_domain: matter
 ha_platforms:
   - binary_sensor
+  - climate
```
Comment (MartinHjelmare) [inline source/_integrations/matter.markdown:20]: Update the category list too.

Comment (marcelveldt) [reply source/_integrations/matter.markdown:20]: adjusted

Comment (MartinHjelmare) [reply source/_integrations/matter.markdown:20]: Commit wasn't pushed?

Comment (marcelveldt) [reply source/_integrations/matter.markdown:20]: oops, forgot to hit the push button and went for lunch :-)

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**